src/FactoryCollection.php

```diff
@@ -23,12 +23,47 @@ final class FactoryCollection implements \IteratorAggregate
 {
     /**
      * @param Factory<T>                      $factory
-     * @phpstan-param \Closure():iterable<Attributes> $items
+     * @phpstan-param \Closure():iterable<Attributes>|iterable<Factory<T> $items
      */
     private function __construct(public readonly Factory $factory, private \Closure $items)
     {
     }
 
+    /**
+     * @internal
+     */
+    public static function accepts(mixed $potentialFactories): bool
+    {
+        if (!is_array($potentialFactories) || count($potentialFactories) === 0 || !array_is_list($potentialFactories)) {
+            return false;
+        }
+
+        if (!$potentialFactories[0] instanceof Factory) {
+            return false;
+        }
+
+        foreach ($potentialFactories as $potentialFactory) {
+            if (!$potentialFactory instanceof Factory
+                || $potentialFactory::class() !== $potentialFactories[0]::class()) {
```

Comment:
I think we can even check for inheritance between classes?

Reply:
Let's leave as-is for now and adjust if someone hits this.